pkg/tasks/queue/pgqueue.go

```diff
@@ -710,22 +710,31 @@ func (p *PgQueue) RemoveAllTasks() error {
 	return nil
 }
 
-func (p *PgQueue) ListenForCancel(ctx context.Context, taskID uuid.UUID, cancelFunc context.CancelFunc) {
+func (p *PgQueue) ListenForCancel(ctx context.Context, taskID uuid.UUID, cancelFunc context.CancelCauseFunc) {
 	logger := zerolog.Ctx(ctx)
 	conn, err := p.Pool.Acquire(ctx)
 	if err != nil {
-		logger.Error().Err(err).Msg("ListenForCancel: error acquiring connection")
+		if !errors.Is(ErrNotRunning, context.Cause(ctx)) {
+			logger.Error().Err(err).Msg("ListenForCancel: error acquiring connection")
+		}
 		return
 	}
 	defer conn.Release()
 
 	// Register a channel for the task where a notification can be sent to cancel the task
 	channelName := getCancelChannelName(taskID)
+	// TODO remove debug logs. put there to see if Exec is getting stuck
+	logger.Debug().Msg("ListenForCancel: preparing listen statement")
 	_, err = conn.Conn().Exec(ctx, "listen "+channelName)
 	if err != nil {
+		//if !errors.Is(ErrNotRunning, context.Cause(ctx)) {
+		//	logger.Error().Err(err).Msg("ListenForCancel: error registering channel")
+		//}
 		logger.Error().Err(err).Msg("ListenForCancel: error registering channel")
+
 		return
 	}
+	logger.Debug().Msg("ListenForCancel: listening on channel")
```

Comment:
We're seeing an issue where the listener fails to register because the context is canceled. This could happen if the task finishes very quickly. This could also happen if this Exec call is hanging and then unblocking when the task is finished (because the context was canceled). 

I left that error in with some debug logs, the idea being to make sure this call the database isn't hanging.

Reply:
removed this for now